sourcegraph/sourcegraph
# Search: Migrate search context picker to the wildcard UI

Closes https://github.com/sourcegraph/sourcegraph/issues/41633

This PR simply migrates Context Picker UI from the dropdown reactstrap component to the new wildcard Combobox UI. This PR should introduce no changes except one below 

| Before | After |
| ------- | ------ |
| <img width="532" alt="Screenshot 2022-09-16 at 17 43 12" src="https://user-images.githubusercontent.com/18492575/190666145-d5f19e42-62a5-47a1-a671-2a74ca97bd63.png"> | <img width="530" alt="Screenshot 2022-09-16 at 17 42 49" src="https://user-images.githubusercontent.com/18492575/190666230-56e87932-6917-47da-8113-99cb38652fd2.png"> |

As you can see prior to this PR search context picker has one visual state for all possible element states. In this PR context picker has different visual states for 
- Element was selected when we open context picker
- Element has been selected with mouse (hover state)
- Element has been selected with keyboard

The problem with original version that you can't say just looking at the UI on which element you cursor is and on which you navigated with keyboard (because the all look the same) In this PR we tried to solve this problem, these new states comes from the wildcard combobox UI so all other Combobox in the product work in the same way with visual states. 

## Test plan
- Go to the main home page
- Click on the context button 
- You should see context picker popover in the right position 
- You should see that input field has autofocus 
- You should be able go through suggestions list with keyboard (arrows)
- If you close context picker with Esc you should see search box with focus
- If you pick one of the suggestion you should see search box with focus



## App preview:

- [Web](https://sg-web-vk-migrate-search-context-picker.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-mjfdnphigw.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
